.github/workflows/sonar.yml

```
@@ -0,0 +1,29 @@
+name: Sonar_Analysis
+on:
+  pull_request: 
+    branches: 
+      - '**'
+  push: 
+    branch: 
+      - 'main'
```

I think this PR should target `develop` and we should also add the `develop` branch here

```suggestion
      - 'main'
      - 'develop'
```

---

This is on `push`. We don't need the `develop` in the `push` part.
It will run on the same code as from the pull_request that was just merged.

Besides, another check will run on the subsequent and inevitable push to `main`/`master` when the extension gets released.